app/components-react/windows/reactive-data-editor/ReactiveDataEditor.tsx

```diff
@@ -0,0 +1,434 @@
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'column',
+        flex: 1,
+        width: '100%',
+        height: '100%',
+        overflow: 'hidden',
+        border: `1px solid ${colors.borderWhite10}`,
+        backgroundColor: colors.dark1,
+      }}
+    >
+      <div
+        style={{
+          borderBottom: `1px solid ${colors.borderWhite10}`,
+          backgroundColor: 'rgba(23, 36, 45, 0.5)',
+          padding: '16px 24px',
+        }}
+      >
+        <h2
+          style={{
+            margin: 0,
+            fontSize: '1.125rem',
+            fontWeight: 600,
+            letterSpacing: '-0.01em',
+            color: '#ffffff',
+          }}
+        >
+          Reactive State
+        </h2>
+        <p
+          style={{
+            marginTop: '4px',
+            fontSize: '0.875rem',
+            color: colors.textWhite60,
+          }}
+        >
+          Manage your reactive state values
+        </p>
+      </div>
+
+      <section
+        style={{
+          display: 'flex',
+          flex: 1,
+          flexDirection: 'column',
+          overflowY: 'auto',
+        }}
+      >
+        <div
+          style={{
+            borderTop: '1px solid transparent',
+          }}
+        >
+          {entries.map(({ key, info }, index) => {
+            const isDirty = dirtyMap[key] ?? false;
+
+            return (
+              <div
+                key={key}
+                style={{
+                  display: 'flex',
+                  flexDirection: 'row',
+                  gap: '12px',
+                  padding: '16px 24px',
+                  transition: 'background-color 0.2s ease',
+                  borderTop: index === 0 ? 'none' : `1px solid ${colors.borderWhite5}`,
+                  alignItems: 'center',
+                  justifyContent: 'space-between',
+                }}
+              >
+                <div
+                  style={{
+                    flex: 1,
+                    display: 'flex',
+                    flexDirection: 'column',
+                    gap: '4px',
+                  }}
+                >
+                  <label
+                    htmlFor={key}
+                    style={{
+                      fontSize: '0.875rem',
+                      fontWeight: 600,
+                      color: isDirty ? colors.streamlabs : colors.textWhite90,
+                      transition: 'color 0.2s ease',
+                    }}
+                  >
+                    {info.name}
+                  </label>
+                  <div
+                    style={{
+                      fontSize: '0.75rem',
+                      color: colors.textWhite60,
+                    }}
+                  >
+                    {key.split('.')[1]}
+                  </div>
+                </div>
+
+                <div
+                  style={{
+                    width: '200px',
+                  }}
+                >
+                  <input
```

Comment:
should probably be using the `NumberInput` component here, makes things a lot easier